### PR TITLE
Fix IndexSizeError DOMException

### PR DESCRIPTION
**Pull request by teoli2003**
`IndexSizeError` is a type of `DOMException`. THis PR makes this clear in all cases.

### CODE REVIEW

File: files/en-us/web/api/audionode/connect/index.md [hunk 1]
```diff
@@ -63,16 +60,15 @@ If the destination is an `AudioParam`, `connect()` returns
...
+- `IndexSizeError` {{domxref("DOMException")}}
+  - : Thrown if the value specified as `outputIndex` or `inputIndex` doesn't correspond to an existing input or output.
+- `InvalidAccessError` {{domxref("DOMException")}}
+  - : Thrown if the destination node is not part of the same audio context as the source node.
+- `NotSupportedError` {{domxref("DOMException")}}
+  - : Thrown if the specified connection would create a cycle (in which the audio loops back through
+    the same nodes repeatedly) and there are no {{domxref("DelayNode")}} objects in the cycle to
     prevent the resulting waveform from getting stuck constructing the same audio frame
-    indefinitely.
+    indefinitely. Also thrown if the `inputIndex` parameter is used whiile the destination is an {{domxref("AudioParam")}}.
```
Comment (wbamberg) [inline files/en-us/web/api/audionode/connect/index.md:71]:
```suggestion
    indefinitely. Also thrown if the `inputIndex` parameter is used while the destination is an {{domxref("AudioParam")}}.
```

File: files/en-us/web/api/audionode/disconnect/index.md [hunk 2]
```diff
@@ -38,18 +38,18 @@ AudioNode.disconnect(destination, output, input);
...
+  - : An index describing which input into the specified destination `AudioNode` is to be disconnected. The index numbers are defined according to the number of input channels (see [Audio channels](/en-US/docs/Web/API/Web_Audio_API/Basic_concepts_behind_Web_Audio_API#audio_channels)).
 
 ### Exceptions
 
-- `IndexSizeError`
-  - : A value specified for `input` or `output` is invalid, referring to a node which doesn't exist or outside the permitted range.
-- `InvalidAccessError`
-  - : The node on which `disconnect()` was called isn't connected to the specified `destination` node.
+- `IndexSizeError` {{domxref("DOMException")}}
+  - : Thrown if the value specified as `input` or `output` is invalid, referring to a node which doesn't exist or outside the permitted range.
```
Comment (wbamberg) [inline files/en-us/web/api/audionode/disconnect/index.md:50]:
```suggestion
  - : Thrown if the value specified as `input` or `output` is invalid, referring to a node which doesn't exist or is outside the permitted range.
```

File: files/en-us/web/api/canvasrenderingcontext2d/createimagedata/index.md [hunk 3]
```diff
@@ -51,17 +51,15 @@ object is filled with transparent black pixels.
 This snippet creates a blank `ImageData` object using the
 `createImageData()` method.
 
-#### HTML
+The HTML is very simple:
```
Comment (wbamberg) [inline files/en-us/web/api/canvasrenderingcontext2d/createimagedata/index.md:54]: I would just omit this line. It is simple, but I never like calling things simple in docs.

File: files/en-us/web/api/canvasrenderingcontext2d/createimagedata/index.md [hunk 4]
```diff
@@ -77,14 +75,12 @@ console.log(imageData);
 
 This example creates and fills a new `ImageData` object with purple pixels.
 
-#### HTML
+The HTML is trivial, consisting only of the `<canvas>` element:
```
Comment (wbamberg) [inline files/en-us/web/api/canvasrenderingcontext2d/createimagedata/index.md:78]: Also here.

File: files/en-us/web/api/canvasrenderingcontext2d/getimagedata/index.md [hunk 5]
```diff
@@ -81,14 +81,12 @@ canvas specified. The coordinates of the rectangle's top-left corner are
 This example draws a rectangle, and then uses `getImageData()` to grab a
 portion of the canvas.
 
-#### HTML
+The HTML is very simple:
```
Comment (wbamberg) [inline files/en-us/web/api/canvasrenderingcontext2d/getimagedata/index.md:84]: ...and again.

File: files/en-us/web/api/htmltableelement/insertrow/index.md [hunk 6]
```diff
@@ -43,15 +43,21 @@ element.
...
 
 ### Return value
 
-`newRow` is an {{domxref("HTMLTableRowElement")}} that references the new
+An {{domxref("HTMLTableRowElement")}} that references the new
 row.
 
+### Exceptions
+
+The following exceptions can be raised:
```
Comment (wbamberg) [inline files/en-us/web/api/htmltableelement/insertrow/index.md:56]: We don;t usually have this preamble, is there a reason for it here?

File: files/en-us/web/api/htmltableelement/insertrow/index.md [hunk 7]
```diff
@@ -43,15 +43,21 @@ element.
...
+
+- `IndexSizeError` {{domxref("DOMException")}}
+  - : Thrown if `index` is greater than the number of rows
```
Comment (wbamberg) [inline files/en-us/web/api/htmltableelement/insertrow/index.md:59]:
```suggestion
  - : Thrown if `index` is greater than the number of rows.
```

File: files/en-us/web/api/range/setend/index.md [hunk 8]
```diff
@@ -30,30 +30,27 @@ range.setEnd(endNode, endOffset);
...
+  - : An integer greater than or equal to zero representing the offset for the end of the `Range` from the start of `endNode`.
 
 ### Return value
 
 `undefined`.
 
 ### Exceptions
 
-Exceptions are thrown as {{domxref("DOMException")}} objects of the following types:
+Possible exceptions are:
```
Comment (wbamberg) [inline files/en-us/web/api/range/setend/index.md:41]: Also here.

File: files/en-us/web/api/range/setend/index.md [hunk 9]
```diff
@@ -30,30 +30,27 @@ range.setEnd(endNode, endOffset);
 - `endNode`
   - : The {{ domxref("Node") }} inside which the {{ domxref("Range") }} should end.
 - `endOffset`
-  - : An integer greater than or equal to zero representing the offset for the end of the
-    `Range` from the start of `endNode`.
+  - : An integer greater than or equal to zero representing the offset for the end of the `Range` from the start of `endNode`.
 
 ### Return value
 
 `undefined`.
```
Comment (wbamberg) [inline files/en-us/web/api/range/setend/index.md:37]: Should we standardise on "None." for this? I like "None." better than `undefined` or Void.

**Review by wbamberg — CHANGES_REQUESTED** [review]
These are nice changes. I had a few editorial suggestions.

File: files/en-us/web/api/range/setend/index.md (hunk 9 above)
Comment (teoli2003) [reply files/en-us/web/api/range/setend/index.md:37]: Yes, I think _None._ is the best way. There are a lot of these: I may script these, should cover 95% of the cases.

Comment (wbamberg) [reply files/en-us/web/api/range/setend/index.md]: Yes, I only mentioned this one because you changed a "Void" down below.

**Review by wbamberg — APPROVED** [review]
👍 thanks @teoli2003 !

**Pull request merged**